streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -58,6 +63,11 @@ public int partition() {
         return partition;
     }
 
+    @Override
+    public Headers headers() {
+        return headers;
+    }
+
     @Override
     public boolean equals(final Object o) {
```

Comment:
Do we need to update `equals` and `hashCode`? (not sure)

Reply:
I think so. I think there is not reason to have the same records, from the same cluster with different headers. Adding it.